app/service/planning_svc.py

```diff
@@ -25,17 +24,22 @@ def __init__(self):
             self.log.debug('Stopping conditions met. No more links will be generated!')
             planner.stopping_condition_met = True
             return []
-        if phase:
-            abilities = [i for p, v in operation.adversary.phases.items() if p <= phase for i in v]
+        if operation.atomic_enabled:
+            if operation.last_ran is None:
+                abilities = [operation.adversary.atomic_ordering[0]]
+            else:
+                abilities = operation.adversary.atomic_ordering[:(operation.adversary.atomic_ordering.index(
+                    operation.last_ran) + 2)]
```

Comment:
This seems kinda hacky. I would think that since we are already tracking the last_ran, we can just trim the list using that without having to having an if statement.  So if an operation initializes last_ran as 0 vs None, could we simplify this?

Reply:
Ah so last_ran is the actual ability object - missed that

Reply:
let's try to clean this up as well

Reply:
agreed. this can be cleaned up a bit with a few minor tweaks syntactically.